monitoring/sdm_health_exporter/sdm_health_exporter.py

```diff
@@ -0,0 +1,178 @@
+import os
+import time
+from typing import Any
+from pydantic import BaseModel
+import strongdm
+from prometheus_client import start_http_server, Gauge
+
+"""
+sdm_exporter.py
+
+This script serves as an example exporter that can monitor the
+health of resources ("Infrastructure") and nodes ("Gateways/Relays").
+
+The script uses the following workflow:
+
+- Make an API call to strongDM's API to retrive information about resources
+and nodes. The frequency of the API call is configurable by updating the
+"update_interval" variable in "main()"
+
+- Collect data about any resource or node that is tagged with "alert" in strongDM.
+This tag is configurable by updating the "alert_tag" variable in "main()"
+
+- Export metrics to a prometheus endpoint as a gauge (0 for healthy, 1 for unhealthy)
+"""
+
+class SdmObject(BaseModel):   
+    """
+    Pydantic class with type enforcement
+    Prometheus object of type "Gauge"
+    Possible values: healthy: 0, unhealthy: 1
+    """
+    health_metric: Gauge
+
+    class Config:
+        validate_assignment = True
+        arbitrary_types_allowed = True
+
+
+def strip_invalid_chars(name):
+    """
+    Dashes are invalid for prometheus metric but allowed in strongDM
+    """
+    return name.replace('-','_')
+
+
+def get_sdm_keys():
+    """
+    Loads API keys from environment variables.
+    Raises: KeyError if environment variables are not found
+    Returns: strings for API ID and secret
+    """
+
+    try:
+        api_id = os.environ['SDM_API_ID']
+        api_secret = os.environ['SDM_API_SECRET']
```

Comment:
Rename:
- SDM_API_ID to SDM_API_ACCESS_KEY
- SDM_API_SECRET to SDM_API_SECRET_KEY